core/src/com/unciv/ui/map/TileGroupMap.kt

```diff
@@ -49,20 +90,44 @@ class TileGroupMap<T: TileGroup>(val tileGroups: Collection<T>, val padding: Flo
             unitImageLayers.add(group.unitImageLayerGroup.apply { setPosition(group.x,group.y) })
             cityButtonLayers.add(group.cityButtonLayerGroup.apply { setPosition(group.x,group.y) })
             circleCrosshairFogLayers.add(group.circleCrosshairFogLayerGroup.apply { setPosition(group.x,group.y) })
+
+            if (worldWrap){
```

Comment:
It seems that a better way to add all tileGroups including the left and right mirrors would be to add a `var tileGroupsToAdd = tileGroups.asSequence()` or something at line 28, and then `if(worldWrap) {tileGroupsToAdd += leftMirrorTileGroups.asSequence(); rightMirrorTileGroups.asSequence() }

That way you can avoid the double-additions both here and in lines 56-71 and 119-122

Reply:
There are two things why I didn't use a sequence. Lines 63 and 71 are not completely the same. L63 is `+ bottomX * 2` and L71 is `- bottomX * 2`. I don't know how I would solve this with a Sequence.
Regarding Lines 95-111: when using a Sequence it would have to get sorted like tileGroups is now
```
 // Apparently the sortedByDescending is kinda memory-intensive because it needs to sort ALL the tiles
        for(group in tileGroups.sortedByDescending { it.tileInfo.position.x + it.tileInfo.position.y }){
```
The advantage of using the hash maps is that I can get the tileInfo of the sorted Collection and render everything in the right order without sorting everything three times see L95 `var group = leftMirrorTileGroups[group.tileInfo]!!`.
I'm shadowing the group val from the for-loop right there out of convenience and could change its name to make it a bit clearer.

I could also use a HashMap<TileInfo, Pair\<T\>> instead of two HashMap<TileInfo, T> to reduce the code size.